mindsdb/integrations/handlers/writer_handler/settings.py

```diff
@@ -42,7 +185,74 @@ class ModelParameters(BaseModel):
     verbose: bool = False
 
     class Config:
+        extra = Extra.forbid
+        arbitrary_types_allowed = True
+
+
+class MissingPromptTemplate(Exception):
+    pass
+
+
+class UnsupportedVectorStore(Exception):
+    pass
+
+
+class MissingUseIndex(Exception):
+    pass
+
+
+class WriterHandlerParameters(BaseModel):
+    """Model parameters for create model"""
+
+    prompt_template: str
+    llm_params: WriterLLMParameters
+    chunk_size: int = 500
+    chunk_overlap: int = 50
+    evaluation: bool = False
+    accuracy_threshold: float = 0.6
+    evaluate_dataset: Union[pd.DataFrame, str] = "squad_v2_val_100_sample"
+    run_embeddings: bool = True
+    external_index_name: str = None
+    top_k: int = 4
+    embeddings_model_name: str = DEFAULT_EMBEDDINGS_MODEL
+    context_columns: Union[List[str], str] = None
+    vector_store_name: str = "chroma"
+    vector_store: VectorStore = VectorStoreFactory.get_vectorstore(vector_store_name)
+    collection_or_index_name: str = "langchain"
+    vector_store_folder_name: str = "chromadb"
+    vector_store_storage_path: str = None
+    evaluation_output: Dict = None
+
+    class Config:
+        extra = Extra.forbid
         arbitrary_types_allowed = True
+        use_enum_values = True
+
+    @validator("vector_store_name")
+    def name_must_be_lower(cls, v):
+        return v.lower()
+
+    @validator("prompt_template")
+    def prompt_template_must_be_provided(cls, v):
+        if not v:
+            raise MissingPromptTemplate(
+                "Please provide a `prompt_template` for this engine."
+            )
+        return v
```

Comment:
Using pydantic BaseModel for user input args in USING is a good way to better validate input parameters are valid format and are not missing. I have also set it up so that it will wrong an error if a user gives an extra param which is currently not supported.